core/pom.xml

```diff
@@ -285,6 +286,18 @@
       <scope>test</scope>
     </dependency>
 
+    <!-- Logging -->
+    <dependency> 
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
+      <version>${slf4j.version}</version>
+    </dependency>
+    <dependency> 
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-jdk14</artifactId>
```

Comment:
Should this be a test only dependency? 

It looks like this is meant for deployment time configuration otherwise: https://www.slf4j.org/manual.html#swapping.

Reply:
I've updated the scope to `runtime`. I think we should have logs to debug issues outside of the test scope. Please let me know what you think.

Reply:
Maybe I've got this wrong, but I'd expect we code to the slf4j-api and then allow applications to supply the "provider". Since we do want logs in test, we should make sure we have a provider in a test scope. Otherwise, we don't want to impose an implementation on our customers.

See https://www.slf4j.org/manual.html#libraries for details.

Reply:
What if we remove `slf4j-jdk14` dependency, and then add a `<scope>test</scope>` dependency for logback-classic-1.4.6.jar. I prefer Logback to Java 1.4 logging because I find it is much easier to configure.

Reply:
I've added a `<scope>test</scope>` dependency for `logback-classic` as suggested.

Note: the last `logback-classic` version that supports Java 8 is `1.3.11`